webknossos/webknossos/dataset/dataset.py

```diff
@@ -506,16 +418,164 @@ def open(cls, dataset_path: str | PathLike, read_only: bool = False) -> "Dataset
         """
 
         dataset_path = strip_trailing_slash(UPath(dataset_path))
-        dataset_properties = cls._load_properties(dataset_path)
+        dataset_properties = cls._load_dataset_properties_from_path(dataset_path)
 
         dataset = cls.__new__(cls)
         dataset.path = dataset_path
-        dataset._read_only = read_only
         dataset._resolved_path = cheap_resolve(dataset_path)
-        return dataset._init_from_properties(dataset_properties)
+        dataset._init_from_properties(dataset_properties, read_only)
+        return dataset
+
+    @classmethod
+    def open_remote(
```

Comment:
Should this and `download` also be moved to `AbstractDataset` as `get_remote_datasets`?